internal/pkg/manifest/validate.go

```diff
@@ -853,21 +853,31 @@ func (ip IPNet) validate() error {
 
 // validate returns nil if NetworkLoadBalancerConfiguration is configured correctly.
 func (c NetworkLoadBalancerConfiguration) validate() error {
-	if err := c.MainListener.validate(); err != nil {
-		return fmt.Errorf("validate nlb.listener; %s", err)
+	if c.IsEmpty() {
+		return nil
+	}
+	if err := c.Listener.validate(); err != nil {
+		return fmt.Errorf("validate nlb: %w", err)
+	}
+	if err := c.Aliases.validate(); err != nil {
+		return fmt.Errorf(`validate nlb: validate "alias": %w`, err)
+	}
+	if !c.Aliases.IsEmpty() {
+		for _, advancedAlias := range c.Aliases.AdvancedAliases {
+			if advancedAlias.HostedZone != nil {
+				return fmt.Errorf(`validate nlb: "hosted_zone" is not supported for Network Load Balancer`)
+			}
+		}
 	}
 	for idx, listener := range c.AdditionalListeners {
 		if err := listener.validate(); err != nil {
-			return fmt.Errorf("validate nlb.listener[%d]; %s", idx, err)
+			return fmt.Errorf("validate nlb.additional_listeners[%d]: %w", idx, err)
```

Comment:
If we remove all `validate "nlb"` wrapping in this `validate()`, then
```suggestion
			return fmt.Errorf(`validate "additional_listeners[%d]": %w`, idx, err)
```